Main.py

```diff
@@ -340,6 +353,59 @@ async def saturdayrank(ctx):
     await _rank(ctx, saturday=True)
 
 
+async def _hist(ctx, saturday: bool = False):
+    conn = sqlite3.connect(DB_PATH)
+
+    try:
+        c = conn.cursor()
+
+        all_scores = _get_times(c)[int(saturday)]
+        scores = _get_times(c, ctx.author)[int(saturday)]
+
+        if not (all_scores and scores):
+            await ctx.send(NO_TIMES_MESSAGE)
+            return
+
+        bins = tuple(range(30, 185, 5)) if saturday else tuple(range(10, 165, 5))
+
+        fig, ax = plt.subplots(nrows=1, ncols=1)
+        ax.set_title(f"{ctx.author.name}'s {'saturday ' if saturday else ''}score histogram",
+                     color="#DCDDDE")
+        ax.set_facecolor("#40444B")
+        ax.hist(all_scores, bins, density=True, color="#942626")
+        ax.hist(scores, bins, density=True, color="#F04747")
+        ax.set_xlabel("time (seconds)", color="#DCDDDE")
+        ax.set_ylabel("density", color="#DCDDDE")
+        ax.tick_params(axis="x", colors="#DCDDDE")
+        ax.tick_params(axis="y", colors="#DCDDDE")
+
+        with io.BytesIO() as tf:
+            fig.savefig(tf, dpi=300, facecolor="#2F3136", edgecolor="#2F292B")  # defaults to PNG
+            tf.seek(0)
+            await ctx.send(file=discord.File(tf, filename="hist.png"))
+
+    finally:
+        conn.close()
+
+
+@bot.command()
+@commands.has_role(CROSSWORD_ROLE)
+async def hist(ctx):
```

Comment:
I think it would be nice if you could call hist on other users. I'll write an issue for the other commands that don't do it since that takes more time to implement.

Reply:
Will be done in the PR for https://github.com/idoneam/minicrossword-bot/issues/38